mtags/src/main/scala/scala/meta/internal/metals/TrigramSubstrings.scala

```diff
@@ -5,7 +5,10 @@ import java.lang.StringBuilder
 /**
  * Returns all possible substrings of length 3 for the given string.
  */
-class TrigramSubstrings(string: String) extends Traversable[String] {
+class TrigramSubstrings(string: String) extends Iterable[String] {
+  // TODO(gabro)
+  override def iterator: Iterator[String] = ???
```

Comment:
I've literally never in my life implemented an `Iterator`, but this shouldn't be hard I presume.

Reply:
@gabro  see this scala 2.13 [write up](https://docs.scala-lang.org/overviews/core/custom-collections.html) by Odersky et al.
```
trait CustomIterable { self =>
  def iterator: Iterator[Foo] = new AbstractIterator[Foo] {
    private var current = 0
    def hasNext = current < self.length
    def next(): Foo = {
      val elem = self(current)
      current += 1
      elem
    }
  }

  def children: List[Foo] // implement in extended classes
  def apply(i: Int) = children(i)
  def length = children.size
}
```